STYLE_GUIDE.md

```diff
@@ -0,0 +1,88 @@
+# Style Guide
+
+This document outlines the style guide for the `epinowcast` package. This guide is a work in progress and will be updated as the package evolves. We welcome contributions to this guide and encourage you to raise issues or submit PRs if you have any suggestions.
+
+In addition to this guide we also follow the [tidyverse style guide](https://style.tidyverse.org/). This guide is a subset of the `tidyverse` style guide and outlines the additional style requirements for the `epinowcast` package.
+
+## Naming conventions
+
+- We use a `enw_` prefix to delineate functions that are part of the `epinowcast` package. This is to avoid conflicts with other packages and to make it clear to users which functions are part of the package.
+- The use of this prefix is not required for internal functions (i.e. functions that are not exported) or that are unlikely to have naming conflicts with other packages.
+
+## Dependencies
+
+In general we aim to minimise dependencies on other packages where possible. This makes it easier to maintain the package and reduces the risk of breaking changes in other packages impacting our users. However, additional dependencies are sometimes necessary to improve the functionality of the package.
+
+The following guidelines should be followed when using adding dependencies:
+
+- Added to the `Imports` or `Suggests` field `DESCRIPTION` file in alphabetical order.
+- In the PR that adds the dependency this should be clearly stated in the PR description along with a justification for the dependency, the number and type of downstream dependencies, and an assessment of the risk of the dependency breaking.
+
+More generally when adding functions from external packages (i.e. even if they are already a dependency) the following should be followed:
+
+- Documented in function documentation using the `@importFrom` tag.
+- Used within functions using the `package::function` format.
+
+## Input types and checking
+
+- We support inputs that are coercible to `data.table` objects using `data.table::as.data.table()`. This includes `data.frame` and `tibble` objects. This should be clearly documentede in the function documentation.
```

Comment:
Not sure about that last sentence overall. I don't think we need to repeat the coerce-able-to guidance - I'd prefer that we simply label inputs as data.frame, outputs as data.table, and have a paragraph somewhere prominent reminding people that both tibbles and data.tables are data.frames, so are suitable inputs.
```suggestion
- We support inputs that are coercible to `data.table` objects using `data.table::as.data.table()`. This includes `data.frame` and `tibble` objects. This should be clearly documented in the function documentation.
```

Reply:
I really like the guidance lining to as.data.table as it indicates to look there to see what it supports. It also means if they ever support more things then we don't need to redox

Reply:
going to disagree for now (though I take your point)! Happy to have a wider discussion.